Query improvements

theodosis1989 opened the pull request:
#### What type of PR is this?

<!--
Choose one of the following labels :
- Improvement
- Bug Fix
-->

#### What does this PR do / why is it needed ?

<!--
Describe change being introduced by this PR.
-->

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

#### Other notes for reviewers:

#### Does this PR introduce a user-facing change?
<!--
-->

theodosis1989 committed:
added getExtraElementForPathToElementRegisters in MongoDBCompilerExte… (#30)

* added getExtraElementForPathToElementRegisters in MongoDBCompilerExtension

* added v1_31_0 protocol

theodosis1989 committed:
WIP - supporting nested levels in comparison queries

theodosis1989 committed:
removed the v1_31_0 protocol pure files

theodosis1989 committed:
nested field for comparison operators, tests

theodosis1989 committed:
Merge remote-tracking branch 'coworker/mongo-execute-p10' into query-improvements

theodosis1989 committed:
naming improvements

theodosis1989 committed:
naming improvements(2)

theodosis1989 committed:
Merge remote-tracking branch 'coworker/mongo-execute-p11' into query-improvements

ghub-real commented on legend-engine-xt-nonrelationalStore-mongodb-pure/src/main/resources/core_nonrelational_mongodb/pureToDatabaseCommand/pureToDatabaseCommand.pure:
```diff
@@ -84,29 +84,40 @@ function meta::external::store::mongodb::functions::pureToDatabaseCommand::proce
 
+function meta::external::store::mongodb::functions::pureToDatabaseCommand::buildDotNotationPath(currentPath: String[1], pathToAppend: String[1]): String[1]
+{
+  if ($currentPath == '', | $pathToAppend, | $pathToAppend + '.' + $currentPath);
```
do we need change the variable names?   We are pre-fixing the "path to append" not appending

theodosis1989 replied:
yes right, this should be pathToPrepend. This is a new method because we had the same logic in two places already, and now it was going in a third one as well.

ghub-real replied:
makes sense, pathtoPrefix

ghub-real commented on legend-engine-xt-nonrelationalStore-mongodb-javaPlatformBinding-pure/src/main/resources/core_nonrelational_mongodb_java_platform_binding/test/testSetup.pure:
```diff
@@ -166,7 +168,7 @@ function meta::external::store::mongodb::executionPlan::platformBinding::legendJ
             meta::pure::mapping::modelToModel::executionPlan::platformBinding::legendJava::inMemoryLegendJavaPlatformBindingExtension(),
             meta::external::shared::format::executionPlan::platformBinding::legendJava::bindingLegendJavaPlatformBindingExtension([
                meta::external::format::json::executionPlan::platformBinding::legendJava::jsonSchemaJavaBindingDescriptor()
```
this is a duplicated function .. there is another place where we define the required extensions (where we updating the MongoOnly extensions for the server)

theodosis1989 replied:
i removed the whole file

ghub-real commented on legend-engine-xt-nonrelationalStore-mongodb-javaPlatformBinding-pure/src/main/resources/core_nonrelational_mongodb_java_platform_binding/test/pureSetup.txt:
```diff
@@ -123,7 +123,7 @@ MongoDBConnection test::mongodb::testConnection
     database: userDatabase;
     store: test::mongodb::mydatabase;
-    serverURLs: [localhost:49829];
+    serverURLs: [localhost:57482];
```
we can probablystop committing this file :) or fix it to 27017 the default mongo port 27017 - ideally you would want the container to map that port)

theodosis1989 committed:
fix: renamed the wrong variable

theodosis1989 committed:
changed test domain model - updated tests

gs-hg committed:
cleanup:

native function to manage mongodb container ( used for tests) in pure MongoDBCommands and MongoTestContainerInvoker ( used to start up container locally for development) now both use same code path and test data.

theodosis1989 committed:
deleted the duplicate setup files for the welcome.pure execution

theodosis1989 committed:
updated the README, kept only one file
